vite.config.js

```diff
@@ -0,0 +1,55 @@
+import { defineConfig } from "vite";
+import pkg from "./package.json" with { type: "json" };
+
+const indexInput = "src/index.ts";
+const tokenInput = "src/token.ts";
+const globalVarName = pkg.name;
+
+export default defineConfig(({ mode }) => {
+  const isCJSBuild = mode === "production";
+
+  return {
+    build: {
+      // for UMD build we do not want to empty directory, so previous builds stay intact
+      emptyOutDir: isCJSBuild,
+      minify: true,
+      sourcemap: true,
+      // UMD build targets lower level ES, while CJS the lowest Node.js LTS version
+      target: isCJSBuild ? "es2022" : "es6",
+      lib: {
+        // leave out token from UMD build
+        entry: isCJSBuild ? [indexInput, tokenInput] : indexInput,
+        name: isCJSBuild ? undefined : globalVarName,
+        formats: isCJSBuild ? ["cjs"] : ["umd"],
+        fileName: (format, entryName) => {
+          switch (format) {
+            case "umd":
+              return `umd/${entryName}.min.js`;
```

Comment:
Why rename it to `.min.js` instead of `.umd.js`?

Reply:
Thinking it suffices that it's placed in a directory called `umd`, so that's why no `*.umd.*` in the name. It's `*.min.*`, because it's getting minified.